server/src/instant/reactive/invalidator.clj

```diff
@@ -332,25 +432,27 @@
   (wal/shutdown! wal-opts))
 
 (defn create-wal-chans []
-  (if-not config/instant-on-instant-app-id
-    (let [chan (a/chan 1 (wal-record-xf))]
-      {:wal-chan chan
-       :close-signal-chan (a/chan)
-       :worker-chan chan})
-    (let [wal-chan (a/chan 1)
-          mult (a/mult wal-chan)
-          worker-chan (a/chan 1 (wal-record-xf))
-          byop-chan (a/chan 1)]
-      (a/tap mult worker-chan)
-      (a/tap mult byop-chan)
-      {:wal-chan wal-chan
-       ;; Nothing will ever be put on this chan,
-       ;; it will be closed when the wal-chan is closed
-       ;; so that the consumer can know to stop waiting for
-       ;; its puts to complete
-       :close-signal-chan (a/chan)
-       :worker-chan worker-chan
-       :byop-chan byop-chan})))
+  (let [users-shims (tracer/with-span! {:name "invalidator/init-users-shims"}
+                      (atom (attr-model/get-all-users-shims aurora/conn-pool)))]
```

Comment:
There is a race condition. I wasn't going to worry about it because it seems unlikely that we'd hit it and I thought it would be a lot of work to fix, but turns out its pretty easy to fix.